### PR TITLE
Add `terminate` ingest processor

### PR DESCRIPTION
**Pull request by PeteGillinElastic**
Adds `terminate` to the ingest processor spec.

Closes elastic/elasticsearch#110218

### CODE REVIEW

File: specification/ingest/_types/Processors.ts [hunk 1]
```diff
@@ -230,6 +230,12 @@ export class ProcessorContainer {
    * @doc_id split-processor
    */
   split?: SplitProcessor
+  /**
+   * Terminates the current ingest pipeline, causing no further processors to be run.
+   * This will normally be executed conditionally, using the `if` option.
+   * @doc_id terminate-processor
```
[the remark this answers is not in the document]
Comment (PeteGillinElastic) [reply specification/ingest/_types/Processors.ts:236]: Done.

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by PeteGillinElastic** [comment]
@pquentin Any other comments before I merge this?

**Review by flobernd — APPROVED** [review]
LGTM!

**Pull request merged**